common/thorhelper/roxiehelper.cpp

```diff
@@ -1107,9 +1102,7 @@ class CHeapSortAlgorithm : public CSortAlgorithm
         eof = false;
         if (inputAlreadySorted)
         {
-            while (sorted.isItem(curIndex))
-                ReleaseRoxieRow(sorted.item(curIndex++));
-            sorted.kill();
+            roxiemem::ReleaseRoxieRows(sorted, curIndex);
```

Comment:
Does it matter that the value if curIndex is different after this change? I suspect would be cleaner to clear it